docs/cookbook/invoices/generating-invoice-after-payment.rst

```diff
@@ -0,0 +1,169 @@
+How to have the Invoice generated after the payment is paid?
+============================================================
+
+.. note::
+
+    This cookbook describes customization of a feature available only with `Sylius/InvoicingPlugin <https://github.com/Sylius/InvoicingPlugin/>`_ installed.
+
+The invoicing plugin lets you generate and download PDF invoices regarding orders. In its default behavior the invoice
+is generated right after the customer creates and places the order.
+In this cookbook, we will describe how to change this behavior, so the invoice will be created right after the order would be paid.
+
+Why would you customize the invoice generation?
+-----------------------------------------------
+
+In the default case the order should be immutable, so any other data should not be changed. But let's say that your shop is customized with some logic
+which is not out of the box. Maybe one of these changes will let you change the order after it is placed?
+In this case, it would be better to have an invoice generated after a particular step (in the case of this cookbook - after the order is paid).
```

Comment:
TBH I am not full convicted about this sentence, because yes, the order is immutable, but later you mention payment which you can update (change the payment state for example). I suggest deleting this part about order customization.